sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeCreator.scala

```diff
@@ -295,9 +225,29 @@ case class CreateNamedStruct(children: Seq[Expression]) extends Expression {
     }
   }
 
+  /**
+   * Returns Aliased [[Expression]]s that could be used to construct a flattened version of this
+   * StructType.
+   */
+  def flatten: Seq[NamedExpression] = valExprs.zip(names).map {
+    case (v, n) => Alias(v, n.toString)()
```

Comment:
this might create Alias(Alias(expr, n1), n2), don't you want to avoid this?
further more, it's quite possible that n1==n2, it's also possible that expr is a named expression with the same name.

is there anything similar to org.apache.spark.sql.Column#named which can be used directly on expressions? perhaps this is a good opportunity to refactor this into Expression, or org.apache.spark.sql.catalyst.expressions (package object)?